apps/dashboard/lib/trpc/routers/authorization/roles/permissions/search-permissions.ts

```diff
@@ -0,0 +1,77 @@
+import { db } from "@/lib/db";
+import { ratelimit, requireWorkspace, t, withRatelimit } from "@/lib/trpc/trpc";
+import { TRPCError } from "@trpc/server";
+import {
+  LIMIT,
+  PermissionsSearchResponse,
+  permissionsSearchPayload,
+  transformPermission,
+} from "./schema-with-helpers";
+
+export const searchRolesPermissions = t.procedure
+  .use(requireWorkspace)
+  .use(withRatelimit(ratelimit.read))
+  .input(permissionsSearchPayload)
+  .output(PermissionsSearchResponse)
+  .query(async ({ ctx, input }) => {
+    const { query } = input;
+    const workspaceId = ctx.workspace.id;
+
+    if (!query.trim()) {
+      throw new TRPCError({
+        code: "BAD_REQUEST",
+        message: "Search query cannot be empty",
+      });
+    }
```

Comment:
nitpick: you can add the trim to the `permissionSearchResponse` helper so you don't have this extra check